addon/components/frost-list-item.js

```diff
@@ -1,22 +1,48 @@
-/* global $ */
 import Ember from 'ember'
-import _ from 'lodash/lodash'
-import FrostList from '../frost-list/component'
+const {Component, $} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {PropTypes} from 'ember-prop-types'
+import _ from 'lodash'
+import FrostList from './frost-list'
 
-export default Ember.Component.extend({
-  classNameBindings: ['isSelected', 'frost-list-item'],
+export default Component.extend({
 
-  initContext: Ember.on('init', function () {
-    this.set('_frostList', this.nearestOfType(FrostList))
-  }),
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+  classNames: ['frost-list-item'],
+  classNameBindings: [
+    'isSelected',
+    'showDetail:is-expanded:is-collapsed'
+  ],
+
+  propTypes: {
+    showDetail: PropTypes.bool
+  },
 
-  isSelected: Ember.computed('model.isSelected', function () {
+  // == Computed Properties =====================================================
+  @readOnly
+  @computed('model.isSelected')
+  isSelected (isSelected) {
     // TODO: Find a better solution for binding the className to the parent
-    let modelIsSelect = this.get('model.isSelected')
-    modelIsSelect ? $(this.get('element')).parent().addClass('is-selected')
-    : $(this.get('element')).parent().removeClass('is-selected')
-    return modelIsSelect
-  }),
+    isSelected ? $(this.get('element')).parent().addClass('is-selected')
+      : $(this.get('element')).parent().removeClass('is-selected')
+    return isSelected
```

Comment:
Play with this and see if we can drive it top-down 

Maybe it's not necessary based on a 1px highlight border, or maybe we solve this via javascript (pass in the selection information of items +1 -1 index from this - like a `nextSibling` `previousSibling`